Integrated user-listen api

gauravbhardwaj7 opened the pull request:
This PR which is having following changes-
1.) Made music model having fields of the user listen api.
2.) Fetched user listens in the listens view from the api.

 Note -  Please Don't consider the files APICall.swift, APIData.swift and Helper.swift , As I was experimenting things in this branch and unfortunately it also got pushed along with my main code.

gauravbhardwaj7 committed:
Integrated listenBrainz user-listen api

akshaaatt commented on Listenbrainz/repository/HomeRepositoryImpl.swift:
```diff
@@ -0,0 +1,23 @@
+class HomeRepositoryImpl: HomeRepository {
+
+    func fetchMusicData() -> AnyPublisher<MusicModel, AFError> {
+        let url = URL(string: "https://api.listenbrainz.org/1/user/gb1307/listens")!
```
we should pass the username as a parameter here. Better to insert the username as a param in `func fetchMusicData()`

gauravbhardwaj7 replied:
done

akshaaatt commented on Listenbrainz/model/musicModel.swift:
```diff
@@ -0,0 +1,135 @@
+//
```
Can you please make every struct a separate file in the model folder? Also, Keep the file name the same as the struct name.

gauravbhardwaj7 replied:
done

akshaaatt reviewed (changes requested):
Thanks a lot for this amazing work! 💯 
Please remove/delete the files you added by mistake from git.

gauravbhardwaj7 committed:
Deleted APICall.swift

gauravbhardwaj7 committed:
Deleted APIData.swift

gauravbhardwaj7 committed:
Deleted Helper.swift

akshaaatt commented on Listenbrainz/UI/Screens/ListensView.swift:
```diff
@@ -8,27 +8,43 @@
     @EnvironmentObject var spotifyManager: SpotifyManager
     @State private var spotifyID: String = "https://open.spotify.com/track/0DHh3p0g7qmNNfNdRNwL6N"
+    
+    let params = ["max_ts":"john", "min_ts":"123456"] as Dictionary<String, String>
```
These need to be made dynamic as well.

gauravbhardwaj7 replied:
done

gauravbhardwaj7 committed:
Made parameters dynamic

gauravbhardwaj7 committed:
Updated Music Model

Made changes in the music model and updated according to the new JSON response.

gauravbhardwaj7 committed:
Made userName Dynamic

Made userName parameter  in the API dynamic.

gauravbhardwaj7 committed:
Made Every Struct a different file

gauravbhardwaj7 committed:
Added Artist in Model

akshaaatt committed:
Fix response and indentation at a few places

akshaaatt reviewed (approved):
Awesome work right here! 💯 
Thank you so much for working on this :)
